lib/protoflow-blocks/src/blocks/sys/read_socket.rs

```diff
@@ -0,0 +1,240 @@
+extern crate std;
+use crate::prelude::{vec, String};
+use crate::{IoBlocks, StdioConfig, StdioError, StdioSystem, System};
+use protoflow_core::{
+    types::Any, Block, BlockError, BlockResult, BlockRuntime, Message, OutputPort, Port,
+    PortResult, SystemBuilding,
+};
+use protoflow_derive::Block;
+use serde::{Deserialize, Serialize};
+use simple_mermaid::mermaid;
+use std::{
+    format,
+    io::Read,
+    net::{TcpListener, TcpStream},
+    string::ToString,
+    sync::{Arc, Mutex, PoisonError},
+};
+use tracing::{error, info};
+/// A block that reads a proto object from a TCP port.
```

Comment:
Leave a newline prior to doc comments (and run `cargo fmt`!)

Reply:
done